### ISSUE TITLE
Makefile2Cmake

### ISSUE THREAD

**Issue by erayaydin** [BTaskaya/BaseLevelComputerKernel#63]
Hello,

Currently, project use "x86_64-elf-g++" for cross-compiling. Thats mean, project support "cross-compiling". I think -for better cross-compiling- we should use "CMake" instead of "Make"

Thank you for this lovely repo :heart:

**Comment by gokhanettin** [comment]
By cross-compiling if you mean to support build in different operating systems including Windows, which is what CMake is good at, it might be little bit more involved than it sounds. Current Makefile-based build system uses Unix specific commands. It is not hard to run them through CMake's `add_custom_command` but the build system will still require a Unix environment anyway.

**Comment by BTaskaya** [comment]
Not unix specific commands they are Posix-compatible commands.  If you run it on a posix compatible  system it will work. But windows is not currently Posix compatible so i can't do anything for windows, sorry :/ Btw you are right, on all circumstances i can't support windows.

**Comment by gokhanettin** [comment]
Yes, posix-compatible better describe what I meant to point out. If you cannot support Windows you are not the one to be sorry but the people who cannot use free software. They still have chances like WSL and msys2, which also comes with a pacman port.

**Comment by BTaskaya** [comment]
@gokhanettin thx for info.
